src/modules/questions/questions.service.ts

```diff
@@ -20,47 +24,116 @@ export class QuestionsService {
     private readonly questionOptionRepository: Repository<QuestionOption>,
     @Inject(CACHE_MANAGER)
     private readonly cacheManager: Cache,
+    private readonly testsService: TestsService,
+    private readonly dataSource: DataSource,
   ) {}
 
   async create(createQuestionDto: CreateQuestionDto, authContext: AuthContext): Promise<Question> {
-    const { options, ...questionData } = createQuestionDto;
+    const { options, testId, sectionId, isCompulsory, ...questionData } = createQuestionDto;
     
     // Validate question data
     this.validateQuestionData(createQuestionDto);
-    
+
     // Validate question options
     this.validateQuestionOptions(createQuestionDto);
-    
+
     // Validate question parameters
     this.validateQuestionParams(createQuestionDto.type, createQuestionDto.params, createQuestionDto.marks);
     
-    const question = this.questionRepository.create({
-      ...questionData,
-      tenantId: authContext.tenantId,
-      organisationId: authContext.organisationId,
-      createdBy: authContext.userId,
-    });
-
-    const savedQuestion = await this.questionRepository.save(question);
+    if (testId && !sectionId) {
+      throw new BadRequestException('sectionId is required when testId is provided');
+    }
+    if (!testId && sectionId) {
+      throw new BadRequestException('testId is required when sectionId is provided');
+    }
 
-    // Create options if provided
-    if (options && options.length > 0) {
-      const questionOptions = options.map(optionData => 
-        this.questionOptionRepository.create({
-          ...optionData,
+    const queryRunner = this.dataSource.createQueryRunner();
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+    try {
+      // Create question
+      const question = this.questionRepository.create({
+        ...questionData,
+        tenantId: authContext.tenantId,
+        organisationId: authContext.organisationId,
+        createdBy: authContext.userId,
+      });
+      const savedQuestion = await queryRunner.manager.save(Question, question);
+      // Create options if provided
+      if (options && options.length > 0) {
+        const questionOptions = options.map(optionData =>
+          this.questionOptionRepository.create({
+            ...optionData,
+            questionId: savedQuestion.questionId,
+            tenantId: authContext.tenantId,
+            organisationId: authContext.organisationId,
+          })
+        );
+        await queryRunner.manager.save(QuestionOption, questionOptions);
+      }
+      // Add question to test if testId and sectionId are provided
+      if (testId && sectionId) {
```

Comment:
@vinayak-patil so question is created before validating the testId or sectionId if the testId and sectionId is provided?

Ideally validating if test and section exists if testId and sectionId are provided, before creating question, this way there wont be unnecessary orphan questions